packages/babel-plugin/src/helpers/getClosuresAndParams.ts

```diff
@@ -0,0 +1,70 @@
+import type { NodePath } from '@babel/core';
+
+import {types as babelTypes} from '@babel/core';
+import { PRAGMA } from '../identifiers.json';
+
+type Identifiers = Set<string>;
+
+type Result = {
+  params: Identifiers,
+  closures: Identifiers,
+};
+
+export default function getClosuresAndParams(arrowPath: NodePath<babelTypes.ArrowFunctionExpression>): Result {
+  const params = getParams(arrowPath);
+  const closures = getClosures(arrowPath, params);
+
+  closures.delete(PRAGMA); // this will be transformed by babel, don't include it as a closure
+  
+  return {
+    params,
+    closures,
+  };
+}
+
+function getParams(arrowPath: NodePath<babelTypes.ArrowFunctionExpression>): Set<string> {
+  const set = new Set<string>(Object.entries(arrowPath.scope.bindings)
+    .filter(([, binding]) => binding.kind === 'param')
+    .map(([name]) => name))
+
+  return set;
+}
+
+function getClosures(arrowPath: NodePath<babelTypes.ArrowFunctionExpression>, paramsSet: Set<string>): Set<string> {
+  const set = new Set<string>();
+
+  arrowPath.get('body').traverse({
+    Identifier(identifierPath) {
+      if (!isClosureIdentifier(arrowPath, paramsSet, identifierPath)) {
+        return;
+      }
+
+      set.add(identifierPath.node.name);
+    }
+  });
+
+  return set;
+}
+
+function isClosureIdentifier(arrowPath: NodePath<babelTypes.ArrowFunctionExpression>, paramsSet: Set<string>, path: NodePath<babelTypes.Identifier>): boolean {
+  const name = path.node.name;
+
+  if (
+    path.parent.type === 'MemberExpression' &&
```

Comment:
I think you also need to account for object keys.

Reply:
Oh true.

```
const fn = ()=>{
  return obj['foo'];
}
```

`obj` is a closure.